src/session_impl.cpp

```diff
@@ -604,26 +598,11 @@ namespace aux {
 		session_log(" done starting session");
 #endif
 
-		apply_settings_pack(pack);
+		// apply all m_settings to this session
+		run_all_updates(*this);
 
-		// call update_* after settings set initialized
-
-#ifndef TORRENT_NO_DEPRECATE
-		update_local_download_rate();
-		update_local_upload_rate();
-#endif
-		update_download_rate();
-		update_upload_rate();
-		update_connections_limit();
-		update_unchoke_limit();
-		update_disk_threads();
-		update_network_threads();
-		update_upnp();
-		update_natpmp();
-		update_lsd();
-		update_dht();
```

Comment:
what would happen in `master`? there is a fix there to avoid double creation of DHT

Reply:
I'm not sure. do you know if there's a test to detect the double-creation?

Reply:
I would expect this to be the only place the DHT be created on startup though

Reply:
yes, there is a test for that